extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -750,6 +752,25 @@ public boolean isAnyFetchActive()
                              .anyMatch(fetch -> (fetch != null && !fetch.isDone()));
   }
 
+  /**
+   * Is costly and requires polling the shard to determine if it's empty
+   * @param stream to which shard belongs
+   * @param shardId of the shard
+   * @return if the shard is empty
+   */
+  public boolean isShardEmpty(String stream, String shardId)
```

Comment:
To be clear, this method will not return true for all the empty shards. it will return true for all the empty shards that are also closed. is that correct?

Reply:
Yes, will rename it to isClosedShardEmpty, like it is in KinesisSupervisor